lua/wikis/lab/Widget/ProjectOverview.lua

```diff
@@ -0,0 +1,128 @@
+
+local Grid = Lua.import('Module:Widget/Grid')
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+local IconFa = Lua.import('Module:Widget/Image/Icon/Fontawesome')
+local IconImage = Lua.import('Module:Widget/Image/Icon/Image')
+local Link = Lua.import('Module:Widget/Basic/Link')
+local Panel = Lua.import('Module:Widget/Panel')
+local Widget = Lua.import('Module:Widget')
+local WidgetUtil = Lua.import('Module:Widget/Util')
+
+---@class LabProjectOverviewParameters
+---@field ProjectName string
+---@field ProjectUrl string
+---@field ProjectImage string?
+---@field ProjectImageDark string?
+
+---@class LabProjectOverview: Widget
+---@field props LabProjectOverviewParameters
+local ProjectOverview = Class.new(Widget)
+
+---@return Widget
+function ProjectOverview:render()
+	assert(Logic.isNotEmpty(self.props.ProjectUrl), '|ProjectUrl= not specified')
+	return Grid.Cell{
+		xs = 12,
+		md = 6,
+		cellContent = Panel{
+			heading = self.props.ProjectName,
+			padding = true,
+			children = Grid.Container{
+				gridCells = WidgetUtil.collect(
+					self:_generateImage(),
+					self:_generateOverview()
+				)
+			}
+		}
+	}
+end
+
+---@private
+---@return Widget?
+function ProjectOverview:_generateImage()
+	if Logic.isEmpty(self.props.ProjectImage) then return end
+	return Grid.Cell{
+		xs = 12,
+		xm = 12,
+		md = 6,
+		cellContent = IconImage{
+			imageLight = self.props.ProjectImage,
+			imageDark = Logic.emptyOr(self.props.ProjectImageDark, self.props.ProjectImage),
+			size = '280x180px',
+			link = self.props.ProjectUrl,
+			alignment = 'center'
+		}
+	}
+end
+
+---@private
+---@return Widget
+function ProjectOverview:_generateOverview()
+	return Grid.Cell{
+		xs = 12,
+		sm = 12,
+		md = 6,
+		cellContent = WidgetUtil.collect(
+			HtmlWidgets.H5{
+				children = {
+					IconFa{iconName = 'projecthome', screenReaderHidden = true},
+					Link{link = self.props.ProjectUrl}
+				}
+			},
+			{
+				IconFa{iconName = 'contributors', screenReaderHidden = true},
+				' Current Contributors: ',
+				mw.ext.LiquipediaDB.lpdb('datapoint', {
+					conditions = tostring(
+						ConditionTree(BooleanOperator.all)
+							:add(ConditionNode(ColumnName('type'), Comparator.eq, 'project contributor'))
+							:add(ConditionNode(ColumnName('name'), Comparator.eq, self.props.ProjectUrl))
+					),
+					query = 'count::pageid'
+				})[1].count_pageid,
+			},
+			HtmlWidgets.Br{},
+			{
+				IconFa{iconName = 'articles', screenReaderHidden = true},
+				' Articles Created: ',
+				mw.getCurrentFrame():callParserFunction('#count_pages_by_prefix', self.props.ProjectUrl),
+			},
+			HtmlWidgets.Br{},
+			{
+				IconFa{iconName = 'lastupdated', screenReaderHidden = true},
+				' Last Update: ',
+				Countdown._create{
+					timestamp = DateExt.readTimestamp(
+						mw.getCurrentFrame():callParserFunction(
+							'#lastupdated_by_prefix',
+							self.props.ProjectUrl .. '/'
+						)
+					),
+					rawdatetime = true
+				},
```

Comment:
would we want to supply a formatted date input too?
(for ones that have disbaled js or have set the `Disable countdown time zone adjustment` option in their preferences)

Reply:
no i mean using DateExt.toCountdownArg on the timestamp you already retrieved
along the lines of
```lua
local timeStamp = DateExt.readTimestamp(
	mw.getCurrentFrame():callParserFunction(
		'#lastupdated_by_prefix',
		self.props.ProjectUrl .. '/'
	)
)
.....
Countdown._create{
	timestamp = timeStamp,
	date = DateExt.toCountdownArg(timeStamp)
	rawdatetime = true
},
```